dash.el

```diff
@@ -2166,8 +2166,42 @@ See `-let' for the description of destructuring mechanism."
       ;; TODO: because inputs to the lambda are evaluated only once,
       ;; -let* need not to create the extra bindings to ensure that.
       ;; We should find a way to optimize that.  Not critical however.
-      `(lambda ,(--map (cadr it) inputs)
-         (-let* ,inputs ,@body))))))
+      `(lambda ,(-map #'cadr inputs)
+         (-let ,inputs ,@body))))))
```

Comment:
Changing this to `-let` would be a breaking change, just leave the `-let*` in place. In my previous review I was referring to the surrounding `let*` which can be safely turned into a `let`.

Reply:
Pushed new revision with this change reverted. But I wonder will this be really a breaking change? I guess the only way to access the previous param is with creating custom destructuring and with lexical binding off.

Reply:
Here's a simple way:

```el
(funcall (-lambda ((x) (&plist x y)) y)
         '(:a)
         '(:b b :a a))
```

But I looked again and it seems `-let` is actually implemented in terms of `-let*`, so the breakage I was thinking of doesn't seem to be possible.